react-components/src/components/RevealCanvas/ViewerContext.tsx

```diff
@@ -2,22 +2,34 @@ import { type DataSourceType, type Cognite3DViewer } from '@cognite/reveal';
 import { createContext, type ReactElement, type ReactNode, useContext, useEffect } from 'react';
 import { type RevealRenderTarget } from '../../architecture/base/renderTarget/RevealRenderTarget';
 import { remove } from 'lodash';
+import { type ComponentFactory } from '../Architecture/Factories/ComponentFactory';
 
-export const ViewerContext = createContext<RevealRenderTarget | null>(null);
+type ViewerContextContent = {
+  renderTarget?: RevealRenderTarget;
+  componentFactory?: ComponentFactory;
+};
+
+export const ViewerContext = createContext<ViewerContextContent | null>(null);
 
 export type ViewerContextProviderProps = {
-  value: RevealRenderTarget | null;
+  renderTarget?: RevealRenderTarget;
+  componentFactory?: ComponentFactory;
   children: ReactNode;
 };
 
 export const ViewerContextProvider = ({
-  value,
+  renderTarget,
+  componentFactory,
   children
 }: ViewerContextProviderProps): ReactElement => {
-  useExposeRenderTargetAndViewerSingletons(value ?? undefined);
-  useAddRenderTargetToWindowList(value ?? undefined);
+  useExposeRenderTargetAndViewerSingletons(renderTarget ?? undefined);
+  useAddRenderTargetToWindowList(renderTarget ?? undefined);
 
-  return <ViewerContext.Provider value={value}>{children}</ViewerContext.Provider>;
+  return (
```

Comment:
After thinking about this a little, is it an option to just construct the `ComponentFactory` here? Then it would always be available in the React-context where `RenderTarget` is available, and we don't have to allow this "partial" creation of the `ViewerContextProvider`? What do you think?